### PR TITLE
[22994] - Proper BiometricDidChange Validation.

### PR DESCRIPTION
**Pull request by AlejandroOrozco**
## Ticket
- [22994] - Proper BiometricDidChange Validation.

## Github Issue
- ISSUE-

## Description
- Store the biometricPolicyState associated with the item in the keychain.
- Validate the current biometricPolicyState against the stored one to verify if there was a change.

## Commit message
-  

## Testing
- [x] Added unit tests
- [x] Ran unit tests successfully
- [ ] Added documentation for public APIs and/or Wiki

### CODE REVIEW

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`94.54%`**
File | Coverage
-----|-----
TwilioVerifySDK/TwilioSecurity/Sources/Biometrics/BiometricError.swift | **`71.43%`**
TwilioVerifySDK/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorage.swift | **`92.21%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by mejiagarcia — APPROVED** [review]

File: TwilioVerifySDK/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorage.swift [hunk 1]
```diff
@@ -134,7 +148,7 @@ extension AuthenticatedSecureStorage: AuthenticatedSecureStorageProvider {
     }
   }
 
-  private func canEvaluatePolicy(for authenticator: Authenticator, success: @escaping EmptySuccessBlock, failure: @escaping ErrorBlock) {
+  private func canEvaluatePolicy(for authenticator: Authenticator, with key: String, success: @escaping EmptySuccessBlock, failure: @escaping ErrorBlock) {
```
Comment (yafuquen) [inline TwilioVerifySDK/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorage.swift:151]: Do we need the key for this method?

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`94.55%`**
File | Coverage
-----|-----
TwilioVerifySDK/TwilioSecurity/Sources/Biometrics/BiometricError.swift | **`71.43%`**
TwilioVerifySDK/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorage.swift | **`91.82%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: TwilioVerifySDK/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorage.swift [hunk 2]
```diff
@@ -84,7 +85,20 @@ extension AuthenticatedSecureStorage: AuthenticatedSecureStorageProvider {
...
+
+      Logger.shared.log(withLevel: .info, message: "Verifying biometrics policy state for key: \(key)")
+      if let biometricPolicyState = self.getBiometricPolicyState(for: key),
+         let evaluatedPolicyDomainState = authenticator.context.evaluatedPolicyDomainState {
+          guard biometricPolicyState == evaluatedPolicyDomainState else {
+              Logger.shared.log(withLevel: .info, message: "User did change biometrics")
+              failure(BiometricError.didChangeBiometrics)
+              return
+          }
+      } else if authenticator.context.evaluatedPolicyDomainState != nil {
```
[the remark this answers is not in the document]
Comment (AlejandroOrozco) [reply TwilioVerifySDK/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorage.swift]: @yafuquen Just removed that code, I made it only for migration paths, but no one is using this feature to store items yet, so there's no need. 👍

File: TwilioVerifySDK/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorage.swift (hunk 1 above)
Comment (AlejandroOrozco) [reply TwilioVerifySDK/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorage.swift]: Removed, Thank you 👍

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`94.57%`**
File | Coverage
-----|-----
TwilioVerifySDK/TwilioSecurity/Sources/Biometrics/BiometricError.swift | **`71.43%`**
TwilioVerifySDK/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorage.swift | **`92.72%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`94.57%`**
File | Coverage
-----|-----
TwilioVerifySDK/TwilioSecurity/Sources/Biometrics/BiometricError.swift | **`71.43%`**
TwilioVerifySDK/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorage.swift | **`92.76%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by yafuquen — APPROVED** [review]

**Pull request merged**